core/common/src/test/java/alluxio/network/TieredIdentityFactoryTest.java

```diff
@@ -105,13 +105,24 @@ public void repeatedScriptKey() throws Exception {
     }
   }
 
+  @Test
+  public void unknownScriptKey() throws Exception {
+    String badOutput = "unknown=x";
+    try {
+      runScriptWithOutput(badOutput);
+      fail("Expected an exception to be thrown");
+    } catch (Exception e) {
+      assertThat(e.getMessage(), containsString("Unrecognized tier: unknown"));
+    }
+  }
+
   @Test
   public void invalidScriptOutput() throws Exception {
-    for (String badOutput : new String[] {"x", "a=b c=d", "a=b,cd", "a=b,abc,x=y"}) {
+    for (String badOutput : new String[] {"x", "a=b c=d", "node=b,cd", "node=b,abc,x=y"}) {
       try {
         runScriptWithOutput(badOutput);
         fail("Expected an exception to be thrown");
-      } catch (RuntimeException e) {
+      } catch (Exception e) {
```

Comment:
Can we use `ExpectedException` in these tests instead of parsing the exception messages?

Reply:
`ExpectedException` can only check a single thrown exception, but this test throws an exception with each iteration of the for loop. I've added it for the other tests though